lib/commands/context.js

```diff
@@ -248,15 +253,37 @@ extensions.listWebFrames = async function listWebFrames (useUrl = true) {
 /**
  * Get the contexts available, with information about the url and title of each
  * webview
+ *
+ * @param {Object} opts - Options set, which can include `waitForWebviewMs` to
+ *                        specify the period to poll for available webviews
  * @returns {Array} List of Context objects
  */
-extensions.mobileGetContexts = async function mobileGetContexts () {
+extensions.mobileGetContexts = async function mobileGetContexts (opts = {}) {
+  const {
+    waitForWebviewMs = 0,
+  } = opts;
+
   const curOpt = this.opts.fullContextList;
+  // `appium-ios-driver#getContexts` returns the full list of contexts
+  // if this option is on
+  this.opts.fullContextList = true;
+
+  const timer = new timing.Timer().start();
   try {
-    // `appium-ios-driver#getContexts` returns the full list of contexts
-    // if this option is on
-    this.opts.fullContextList = true;
-    return await this.getContexts();
+    if (waitForWebviewMs === 0) {
+      return await this.getContexts();
+    }
+
+    while (timer.getDuration().asMilliSeconds < waitForWebviewMs) {
```

Comment:
I'd rather use `waitForCondition`

Reply:
That would require an external variable anyway, and be annoying. Also exceptions for control flow are... unneeded.

Reply:
ok, we anyway need to call `getContexts()` at least once.
How about:

```
     let contexts = await this.getContexts();

     while (timer.getDuration().asMilliSeconds < waitForWebviewMs) {
       log.debug('Retrieving contexts');
       contexts = await this.getContexts();
       if (contexts.length >= 2) {
         log.debug(`Found webview context after ${timer.getDuration().asMilliSeconds.toFixed(0)}ms`);
         return contexts;
       }
       log.debug(`No webviews found in ${timer.getDuration().asMilliSeconds.toFixed(0)}ms`);
     }
     return contexts;
```